xcc/settings.py

```diff
@@ -87,14 +94,42 @@ class Config:  # pylint: disable=missing-class-docstring
         env_file = get_path_to_env_file()
         env_prefix = get_name_of_env_var()
 
+    def _check_for_invalid_values(self, key: str, val: str) -> None:
+        """
+        Check for conditions that make saving the env_file
+        dangerous to the user.
+
+        - REFRESH_TOKEN must not contain characters outside Base64URL set
+
+        Args:
+            key (str): .env file key
+            val (str): .env file value
+
+        Raises:
+            Exception: if the value should not be saved to the .env file
+
+        """
+
+        if (
+            key == "REFRESH_TOKEN"
+            and val is not None
+            and re.search(_BASE64URLRE, val) is not None
+        ):
+            raise ValueError("REFRESH_TOKEN contains non-Base64URL character(s)")
```

Comment:
It could make more sense to use "JWT" instead of "Base64URL" given that we're checking for  `.`s too:
```suggestion
            raise ValueError("REFRESH_TOKEN contains non-JWT character(s)")
```